nyaa/api_handler.py

```diff
@@ -186,3 +190,63 @@ def v2_api_upload():
         # Map errors back from form fields into the api keys
         mapped_errors = {UPLOAD_API_FORM_KEYMAP.get(k, k): v for k, v in upload_form.errors.items()}
         return flask.jsonify({'errors': mapped_errors}), 400
+
+
+# ####################################### INFO #######################################
+ID_PATTERN = '^[1-9][0-9]*$'
+INFO_HASH_PATTERN = '^[0-9a-fA-F]{40}$'  # INFO_HASH as string
+
+
+@api_blueprint.route('/v2/info/<torrent_id_or_hash>', methods=['GET'])
+@basic_auth_user
+@api_require_user
+def v2_api_info(torrent_id_or_hash):
+    torrent = None
+    viewer = flask.g.user
+
+    matchID = re.match(ID_PATTERN, torrent_id_or_hash)
+    matchHASH = re.match(INFO_HASH_PATTERN, torrent_id_or_hash)
+
+    if matchID:
+        torrent = models.Torrent.by_id(torrent_id_or_hash)
+    elif matchHASH:
+        # Convert the string representation of a torrent hash into binary
+        a2b_hash = binascii.unhexlify(torrent_id_or_hash)
+        torrent = models.Torrent.by_info_hash(a2b_hash)
+    else:
+        return flask.jsonify({'errors': ['Query was not a valid id or hash.']}), 400
+
+    if not torrent:
+        return flask.jsonify({'errors': ['Query was not a valid id or hash.']}), 400
+
+    # Only allow admins see deleted torrents
+    if torrent.deleted and not (viewer and viewer.is_superadmin):
+        return flask.jsonify({'errors': ['Query was not a valid id or hash.']}), 400
+
+    submitter = 'anonymous'
```

Comment:
Since this is a name, I'd make it `Anonymous` - or possibly, since /user/Anonymous can exist, None